src/snowflake/snowpark/dataframe_stat_functions.py

```diff
@@ -388,6 +389,9 @@ def sample_by(
             col: The name of the column that defines the strata.
             fractions: A ``dict`` that specifies the fraction to use for the sample for each stratum.
                 If a stratum is not specified in the ``dict``, the method uses 0 as the fraction.
+            seed: Specifies a seed value to make the sampling deterministic. Can be any integer between 0 and 2147483647 inclusive.
```

Comment:
what would happen if seed is not in the accepted range?

Reply:
It will raise a SQL error. I think we don't need to do it right now, as `DataFrame.sample` also doesn't do it.